### ISSUE TITLE
Allow Committee Administrator to remove users from Committee Account

### ISSUE THREAD

**Issue by AureliaKhorsand** [fecgov/fecfile-web-app#91]
Business Reason

As a Committee Administrator, I will be able to delete a user from my Committee user list so that I can ensure that the correct users are on my committee access list.

Acceptance Criteria

***Given*** the user is a Committee Administrator

  ***When*** they access the Manage Users Page

  ***Then*** they will be able to select 'Delete' on the report Action list next to a user in the list in any status (Active or Pending)

  ***And when*** the user selects delete

  ***then*** they will see a confirmation message (see wireframe) where the user will be able to:

*   Cancel (nothing happens)
*   Confirm (and the user will be removed from the Manage user list and no longer have access to that committee)

***And*** the user will be unable to delete themselves (there will be no "delete" action button).

Dev Notes

*   This is a hard delete for now.
*   Since you should not be able to delete yourself, please grey out action kabob because there will be no actions in the kabob.

Wireframes

1.  Select a user to delete

       

1.  Confirm delete request

       

1.  Show updated contacts list

       

### QA Notes ###
- [x] BA/Design Quality Checklist complete
- Note that a user can be re-added after they are removed. We should not prevent that.

### DEV Notes ###
null

### Design ###
null

 See full ticket and images here: [FECFILE-1315](https://fecgov.atlassian.net/browse/FECFILE-1315)

**Comment by AureliaKhorsand** [comment]
2/24/22 Requirements Session: Users will only have one role at a time, since permissions are cascading. Users may have different roles per Committee Account they have access to, but NOT within one.

**Comment by MitchellTCG** [comment]
@mary-tcg Please work on a wireframe for this today

**Comment by AureliaKhorsand** [comment]
During [3/17/22 Requirements meeting](https://docs.google.com/document/d/1tQ9GMhrqTDHsVHlIzKxydHEnPT1vm1fShgN-z6Jbfw8/edit#), noted that we will proceed with the term "Backup Committee Administrator" for now, though this may change at a later date. There is no current limit to the number of Backup Committee Administrators, and the role should have the exact same permissions as Committee Administrator (which is all).

**Comment by MitchellTCG** [comment]
Wires updated. UX team will choose a direction in the morning and final wireframes will be added to this ticket.

**Comment by AureliaKhorsand** [comment]
@MitchellTCG @mary-tcg this ticket is ready for a final wireframe, although it's blocked by our login process at the moment so feel free to work on other more pressing wireframes first.

**Comment by AureliaKhorsand** [comment]
Tech Review 4/4/22: Shift paradigm to sending invitations to users, not adding them. Users should be "pending" once an invite is sent, and won't be registered as active users until they successfully log in for the first time through login.gov. This way, we know that for registered users we always have a unique user ID tied to a login token.

**Comment by MitchellTCG** [comment]
![Manage_Contacts_v6.png](https://images.zenhubusercontent.com/619e83beec2d9fadb74ad6d1/c148aa65-d421-4422-b0fc-afcc96bf2b29)

**Comment by MitchellTCG** [comment]
https://images.zenhubusercontent.com/306118986/be839397-8530-4fcf-8739-d26082db2ff8/add_new_user_2_committee_v3.mp4

**Comment by AureliaKhorsand** [comment]
**4/6/22 Refinement Meeting:** Removed "Telephone" from wireframe.

The validation rules for name and email should match Contact field validation rules. 

This ticket does NOT include sending an email to the user being added. That is covered in a separate ticket (Aurelia to add a link to the ticket here).

**Comment by MitchellTCG** [comment]
@AureliaKhorsand @JonellaCulmer 
Wires added

**Comment by JonellaCulmer** [comment]
A couple questions, things for confirmation.

> Users should be "pending" once an invite is sent, and won't be registered as active users until they successfully log in for the first time through login.gov. This way, we know that for registered users we always have a unique user ID tied to a login token.

So it sounds like users will always be in one of two states: Registered and Pending. Does your above explanation mean that a person who belongs to more than one committee account would automatically show as Registered once added by an administrator?

Wondering if we can make the statuses closer to: Active, Inactive and Pending

**Comment by AureliaKhorsand** [comment]
@MitchellTCG @JonellaCulmer this ticket was not marked as ready for wires, so it's still very much up in the air what direction we go in. I also see two sets of wires - can we get rid of the older one?

I like the idea of Active, Inactive, and Pending as statuses. Also not sure about the word "delete" for removing a user, but defer to you all. They'd be rendering the user inactive, not deleting them, right? Is there a better word for that?

**Comment by JonellaCulmer** [comment]
@MitchellTCG @AureliaKhorsand If we're not going to delete users, we can "deactivate" them. We'd also need to change the page title from what it currently is or create a separate table for users who are no longer active or current. 

Perhaps whether we delete or deactivate could be a point of discussion in a future requirements session unless it's already been decided.

**Comment by MitchellTCG** [comment]
@GreggMoreland 
Please review. Removing a user is a hard delete. But the committee administrator can re-invite the user if they want to.

**Comment by mjtravers** [comment]
Placing ticket on hold until #1653 is moved to CR.

**Comment by mjtravers** [comment]
Passes CR. Sending to QA.

**Comment by MitchellTCG** [comment]
### QA verified the following:
**1.** Go to the "Users" page
**2.** Select the action kabob of the user you would like to remove from your committee
**3.** Click delete
**4.** Confirm the deletion
**5.** See the Confirmation toast

<img width="391" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMEdlQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--769e363cdb103868abe87333f65688dceaa3d6da/91%20-Allow%20Committee%20Administrator%20to%20remove%20users%20from%20Committee%20Account.png" alt="91 -Allow Committee Administrator to remove users from Committee Account.png" />

### QA Review Completed. Moved to Stage Ready.
